framework/decode/dx12_dump_resources.cpp

```diff
@@ -2556,8 +2564,8 @@ void DefaultDx12DumpResourcesDelegate::WriteResource(nlohmann::ordered_json&   j
     util::FieldToJson(jdata["res_id"], resource_data->source_resource_id, json_options_);
     util::FieldToJson(jdata["dimension"], util::ToString(resource_data->desc.Dimension), json_options_);
 
-    std::string suffix    = Dx12DumpResourcePosToString(resource_data->dump_position);
-    std::string json_path = suffix + "_file";
+    std::string suffix         = Dx12DumpResourcePosToString(resource_data->dump_position);
+    std::string json_path      = (suffix == "" ? "file" : (suffix + "_file"));
```

Comment:
This looks like a spot fix and not part of "dump resources only after draw calls".  Or is this a rebase issue?

Reply:
Same question for other instances of filename and suffix processing below.

Reply:
See my comments above regarding including "after" and "before". The strange expression is so that we can include an "_" to the file name when "after" or "before" is included in the name, but omit it otherwise. This makes the filenames consistent with Vulkan before/after filenames.

Reply:
👍